### DISCUSSION TITLE
Metaprogramming, insert string into math mode.

### DISCUSSION THREAD

**Discussion by toihr** [typst/typst#1277] [Q&A]
Hi everyone I am just trying to figure out a few things for writing my own scripts and i was wondering if there is a possibility to insert a string aka "a/b" into mathmode such that it displays $\frac{a}{b}$. Is this a thing that is implemented so far?

Kind regards
Tom

**Comment by Enivex** [comment]
You can achieve this with https://typst.app/docs/reference/foundations/eval/

**Reply by toihr** [reply]
Is it intended functionality for eval to not be able to be passed a variable of type string?

**Reply by laurmaedje** [reply]
What do you mean? This works:
```
#let x = "1 + 1"
#eval(x)
```
To create math content, add dollar signs:
```
#let x = "1/2"
#eval("$" + x + "$")
```
Are you sure though that you really need this? Why is the equation a string in the first place? Note that you can also store normal equations in variables:
```
#let x = ($1/2$, $3/4$)
$#x.at(0) + #x.at(1)$
```

**Reply by toihr** [reply]
I am trying to make something akin to siunitx for easy markup of units and for that I need to have access to Math mode formating. And i want to parse it similar to the unit parsing in the python package pint. which basically lets you do u("kg/m").

**Reply by toihr** [reply]
but thank you for iterating on the eval answer, i must have done something weird or maybe i was just tired because I could have sworn that eval did not work for me yesterday.
